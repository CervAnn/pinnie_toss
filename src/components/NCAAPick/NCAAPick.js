import React from 'react'
import { connect } from 'react-redux'
import '../NCAAPick/NCAAPick.css'

export const NCAAPick = ({randomTeam}) => {
  return (
    <section className="NCAAPick">
      <article id="team-name-badge">
        <img id="ncaa-team-badge" src={randomTeam.strTeamBadge} alt=""/>
        <h1>{randomTeam.strTeam} {randomTeam.strAlternate}</h1>
      </article>
    </section>
  )
}

export const mapStateToProps = store => ({
  randomTeam: store.randomTeam
})

export default connect(mapStateToProps)(NCAAPick);