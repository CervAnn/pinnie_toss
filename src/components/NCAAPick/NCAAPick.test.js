import React from 'react';
import { shallow } from 'enzyme';
import { NCAAPick, mapStateToProps } from './NCAAPick';

describe('NCAAPick', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NCAAPick
        teams={[]}
        randomTeam={{
          idTeam: "136915",
          idLeague: "4479",
          strTeam: "Michigan",
          strAlternate: "Wolverines",
          strTeamBadge: "https://www.thesportsdb.com/images/media/team/badge/8f2j7k1564336246.png"
        }}
      />
    );
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })  
  
  describe('mapStateToProps', () => {
    it('should return an object with the contents of the store', () => {
      let mockStore = {
        teams: [
          {
            idTeam: "136915",
            idLeague: "4479",
            strTeam: "Michigan",
            strAlternate: "Wolverines",
            strTeamBadge: "https://www.thesportsdb.com/images/media/team/badge/8f2j7k1564336246.png"
          },
          {
            idTeam: "136916",      
            idLeague: "4479",
            strTeam: "Michigan State",
            strAlternate: "Spartans",
            strTeamBadge: "https://www.thesportsdb.com/images/media/team/badge/2xn7yd1564336256.png"
          }
        ],
        randomTeam: {
            idTeam: "136915",
            idLeague: "4479",
            strTeam: "Michigan",
            strAlternate: "Wolverines",
            strTeamBadge: "https://www.thesportsdb.com/images/media/team/badge/8f2j7k1564336246.png"
        }
      }

      let expected = {
        randomTeam: {
          idTeam: "136915",
          idLeague: "4479",
          strTeam: "Michigan",
          strAlternate: "Wolverines",
          strTeamBadge: "https://www.thesportsdb.com/images/media/team/badge/8f2j7k1564336246.png"
        }
      }

      expect(mapStateToProps(mockStore)).toEqual(expected)
    })
  })
}) 