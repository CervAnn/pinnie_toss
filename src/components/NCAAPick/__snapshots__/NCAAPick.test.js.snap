// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NCAAPick should render without crashing 1`] = `
<section
  className="NCAAPick"
>
  <article
    id="team-name-badge"
  >
    <img
      alt=""
      id="ncaa-team-badge"
      src="https://www.thesportsdb.com/images/media/team/badge/8f2j7k1564336246.png"
    />
    <h1>
      Michigan
       
      Wolverines
    </h1>
  </article>
</section>
`;
